app/src/main/java/com/concordium/wallet/data/preferences/AuthPreferences.kt

```diff
@@ -142,17 +146,90 @@ class AuthPreferences(val context: Context) :
         setString(SEED_PHRASE, "")
     }
 
-    fun setSeedPhrase(value: String) {
+    suspend fun tryToSetEncryptedSeedPhrase(value: String, password: String): Boolean {
         val seed = Mnemonics.MnemonicCode(value).toSeed()
         val seedEncoded = seed.toHex()
-        setString(SEED_PHRASE, seedEncoded)
+        val encryptedSeed = App.appCore.getCurrentAuthenticationManager().encryptInBackground(password, seedEncoded)
+            ?: return false
+       return setStringWithResult(SEED_PHRASE_ENCRYPTED, encryptedSeed)
     }
 
-    fun getSeedPhrase(): String {
-        return getString(SEED_PHRASE, "")
+    suspend fun getSeedPhrase(password: String): String {
+
+        val seedEncrypted = getString(SEED_PHRASE_ENCRYPTED)
+        val seedUnencrypted = getString(SEED_PHRASE)
+
+        if(seedEncrypted != null && seedUnencrypted == null){
+            //Seed is encrypted and the old seed is deleted, decrypt and return the result
+            return App.appCore.getCurrentAuthenticationManager().decryptInBackground(password, seedEncrypted)
+                ?: return ""
+        }
+
+        if(seedEncrypted != null && seedUnencrypted != null){
+            //Booth encrypted and decrypted seeds present, delete the old seed
+            setString(SEED_PHRASE, null)
+            return seedUnencrypted
+        }
```

Comment:
**tryToSetEncryptedSeedPhrase**: this is when we have a new install we pass the seed phrase and the password and we try to save it.
**setSeedPhraseEncrypted**: this is used when we change the password and we have a new encrypted seed
**checkAndTryToEncryptSeed**: this is the main check on the passcode screen, if the seed is unencrypted it will encrypt it and delete the old seed. 

**getSeedPhrase** returns the seed when we are using a password
**getSeedPhraseDecrypted** returns the seed when we are using a SecretKey